### PR TITLE
Make POSIX compliant de/activate scripts

### LINKED ISSUE #50: Activation script is not POSIX compliant, uses bashisms
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

The activate.sh script contains the `[[` bash command when doing a test. This command is not available in `sh`. In this instance, the test can be replaced with `[ -n "$UDUNITS2_XML_PATH" ]` with exactly the same effect, but will work in non-bash shells.

https://github.com/conda-forge/udunits2-feedstock/blob/41a9d87705067782a616960baa4b55113887c98f/recipe/scripts/activate.sh#L6-L8

For context I am running tests for my package using [tox-conda](https://github.com/tox-dev/tox-conda). This writes the output of `conda shell.posix activate [envpath]` plus the test commands to a temporary file, then executes it with `/bin/sh` ([source](https://github.com/AntoineD/tox-conda/blob/6a1f3e937da39d86f4a9d3e0759b9cd724c685b1/tox_conda/env_activator.py#L38-L56)). When running the test command in this fashion the output includes an error message:

```
/tmp/tmp72fqklrw: 6: /path/to/env/.tox/lint/etc/conda/activate.d/udunits2-activate.sh: [[: not found
```

 As the `conda` command explicitly targets `shell.posix` not `shell.bash`, I feel that using POSIX compliant activation scripts is the preferred solution.

### Installed packages

```shell
n/a
```

### Environment info

```shell
n/a
```

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Doing my best to duplicate https://github.com/conda-forge/gdal-feedstock/pull/785

closes #50

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
@mx-moth, could you take a careful look when you have a chance and make sure this should work?  The activate script will be tested with CI but the deactivate script won't, at least in my past experience.

**Comment by xylar** [comment]
@conda-forge/udunits2, I believe this is ready to merge but wouldn't mind another pair of eyes on it.

File: recipe/scripts/activate.sh [hunk 1]
```diff
@@ -1,16 +1,16 @@
-#!/usr/bin/env bash
+#!/usr/bin/env sh
 
 # Store existing env vars and set to this conda env
 # so other installs don't pollute the environment.
 
-if [[ -n "$UDUNITS2_XML_PATH" ]]; then
-    export _CONDA_SET_UDUNITS2_XML_PATH=$UDUNITS2_XML_PATH
+if [ -n "${UDUNITS2_XML_PATH:-}" ]; then
```
[the remark this answers is not in the document]
Comment (mx-moth) [reply recipe/scripts/activate.sh:6]: I can confirm this change works for me, the error message is no longer printed.

[the remark this answers is not in the document]
Comment (mx-moth) [reply recipe/scripts/activate.sh:6]: Adding `:-` to the variables is unnecessary, but doesn't hurt. Using `:-` allows scripts to indicate a default value if the variable is not set. `${FOO}` will return the empty string if $`FOO` is not set, and `${FOO:-}` will return the default value of the empty string if `$FOO` is not set. They are equivalent in this instance. This construct could be useful if the shell is configured to throw an error if a variable is not set, such as `set -o nounset` / `set -u` in bash.

**Review by mx-moth — APPROVED** [review]

**Comment by xylar** [comment]
Thanks @mx-moth for having a look and the clarification. I was following the changes made on gdal so I wasn't at all sure what the purpose of the `:-` was. Good to know that they're unnecessary but harmless.

**Comment by mx-moth** [comment]
Thanks for fixing this!

Reading my comment again, I missed a key example in there. `${FOO:-bar}` will return the contents of `$FOO` if `$FOO` is set and not empty, and will return "bar" if $FOO is unset or is the empty string. The [Bash docs have more comprehensive information](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html).

**Pull request merged**